lib/crypto/src/arithmetic/uint.rs

```diff
@@ -789,7 +789,18 @@ const fn parse_digit(utf8_digit: u8, digit_radix: u32) -> u32 {
     }
 }
 
-/// Parse a single UTF-8 byte.
+/// Parse a single UTF-8 byte into a char.
+///
+/// Converts bytes to characters during compile-time string evaluation.
+/// Only handles ASCII bytes (0x00-0x7F) and panics on non-ASCII bytes.
+///
+/// # Arguments
+///
+/// * `byte` - Byte to convert
+///
+/// # Panics
+///
+/// On non-ASCII bytes (>= 0x80)
```

Comment:
```suggestion
/// Parse a single UTF-8 byte into a char.
///
/// Converts bytes to characters during compile-time string evaluation.
/// Only handles ASCII bytes (0x00-0x7F).
///
/// # Arguments
///
/// * `byte` - Byte to convert.
///
/// # Panics
///
/// * If the byte is non-ASCII (>= 0x80).
```
nit: existing docs convention - use lists where possible, all list items end with dots.

You don't have to spend too much time improving existing documentation for `pub(crate)` and private items, for those it should be enough to have a clear sentence explaining the purpose of the item. 
On the other hand, `pub` items should be properly documented, as you've done here :+1: